golem-worker-service/src/api/openapi/converter.rs

```diff
@@ -0,0 +1,1019 @@
+                    style: Some("form".to_string()),
+                    explode: Some(true),
+                    required: Some(false),
+                    description: None,
+                }
+            );
+        }
+         if route.path.contains("/api/deployments") && route.method == HttpMethod::Get {
+             params.push(
+                Parameter {
+                    name: "api-definition-id".to_string(),
+                    r#in: ParameterLocation::Query,
+                     schema: Schema::String { format: None, enum_values: None },
+                    style: Some("form".to_string()),
+                    explode: Some(true),
+                    required: Some(true),
+                    description: None,
+                }
+            );
+        }
+
+        if route.path.contains("/upload") {
+            params.push(
+                Parameter {
+                    name: "component_type".to_string(),
+                    r#in: ParameterLocation::Query,
+                    schema: Schema::Ref {
+                        reference: "#/components/schemas/ComponentType".to_string()
+                     },
+                    style: Some("form".to_string()),
+                    explode: Some(true),
+                    required: Some(false),
+                    description: Some(
+                        "Type of the new version of the component - if not specified, the type of the previous version is used.".to_string()
+                    ),
+                }
+            );
+        }
+        if route.path.contains("/plugins") && route.method == HttpMethod::Get {
+              params.push(
+                Parameter {
+                    name: "scope".to_string(),
+                    r#in: ParameterLocation::Query,
+                     schema: Schema::Ref {
+                        reference: "#/components/schemas/DefaultPluginScope".to_string()
+                     },
+                    style: Some("form".to_string()),
+                    explode: Some(true),
+                    required: Some(false),
+                    description: None,
+                }
+            );
+         }
+         if route.path.contains("/activate-plugin") || route.path.contains("/deactivate-plugin"){
+             params.push(
+                Parameter {
+                    name: "plugin-installation-id".to_string(),
+                    r#in: ParameterLocation::Query,
+                     schema: Schema::String { format: Some("uuid".to_string()), enum_values: None },
+                    style: Some("form".to_string()),
+                    explode: Some(true),
+                    required: Some(true),
+                    description: None,
+                }
+            );
+         }
+         params
+    }
+
+
+   fn extract_header_parameters(route: &Route) -> Vec<Parameter> {
+        let mut params = Vec::new();
+        if route.path.contains("/invoke-and-await") || route.path.contains("/invoke") {
```

Comment:
I am convinced all this special-case logic around the Golem REST APIs is not correct and not satisfying the requirements of the ticket, which are to convert the user-defined API into OpenAPI schema.